backend/models/plan.py

```diff
@@ -1,4 +1,118 @@
-class plan:
+import datetime
 
-    def __init__(self, name:str, description:str):
-        pass
+from db import db
+from models.user import User
+from exceptions.plan_errors import PlanCreationError, PlanNotFoundError, PlanDBAddingError
+#from google.cloud import firestore
+
+
+class Plan:
+    def __init__(self, name: str, description: str, date: datetime, location: str, admin: User,
+                 participants: list[User], uid: str):
+        self.uid = uid
+        self.name = name
+        self.description = description
+        self.admin: User = admin  # aka the creator
+        self.date: datetime = date
+        self.location: str = location
+        self.participants: list[User] = [admin] if participants is None else participants
+
+    def json(self):
+        return {
+            'id': self.uid,
+            'name': self.name,
+            'description': self.description,
+            'date': self.date.strftime('%d/%m/%Y'),
+            'location': self.location,
+            #'admin': self.admin.json(),
+            'admin': self.admin,
+            'participants': self.participants
+            #'participants': [p.json() for p in self.participants]
```

Comment:
Are the comments relevants? If not, remove them pls

Reply:
idk which option would be better. will explain it in more detail at the weekly meeting